rmw_fastrtps_shared_cpp/src/rmw_count.cpp

```diff
@@ -54,12 +55,24 @@ __rmw_count_publishers(
   WriterInfo * slave_target = impl->secondaryPubListener;
   slave_target->mapmutex.lock();
   *count = 0;
-  for (const auto & it : slave_target->topicNtypes) {
-    const auto topic_fqdn = _demangle_if_ros_topic(it.first);
-    if (topic_fqdn == topic_name) {
-      *count += it.second.size();
+
+  // Build the list of all possible topic FQDN
+  std::vector<std::string> topic_fqdns;
+  topic_fqdns.push_back(topic_name);
+  if (topic_name[0] == '/') {
+    std::for_each(_ros_prefixes.begin(), _ros_prefixes.end(),
```

Comment:
Accessing `_ros_prefixes` while it starts with an underscore seems to be not a good idea since it is considered private. Maybe either change the declaration or expose it through a getter.

Reply:
Sure, I can make a getter. That being said, the original code read: `_demangle_if_ros_topic(it.first)`. Notice the leading `_` character?